src/tests/ftest/util/cmocka_utils.py

```diff
@@ -30,7 +31,8 @@ def get_cmocka_command(command, parameters=None):
                 "buffer overflow detected"]
     path, executable = os.path.split(command)
     command = CmockaCommand(path, executable, keywords)
-    command.parameters.value = parameters
+    if parameters:
+        command.parameters.value = parameters.strip()
```

Comment:
Confirmed that this removes the extra space in the command string used by pgrep/pkill.

An alternative solution would be to ensure that we did not pass in a string with a trailing space via:
```
parameters = [f"-n {dmg_config_file}", f"-{subtest}"]
if args:
    parameters.append(str(args))
daos_test_cmd = get_cmocka_command(command, " ".join(parameters))
```
in `run_subtest()`.

Reply:
I think it makes sense to only append args if non-empty
Maybe the typical `' '.join(filter(None, ...))`?

Reply:
Done